Tasks/DotNetCoreInstallerV0/dotnetcoreinstaller.ts

```diff
@@ -3,28 +3,80 @@ import * as toolLib from 'vsts-task-tool-lib/tool';
 import * as trm from 'vsts-task-lib/toolrunner';
 import { DotNetCoreReleaseFetcher } from "./releasesfetcher";
 import * as utilities from "./utilities";
-
+import * as fileSystem from "fs";
 import * as os from 'os';
 import * as path from 'path';
 
 class DotnetCoreInstaller {
-    constructor(packageType, version) {
+    constructor(workingDirectory: string, packageType: string, version?: string, useGlobalJson?: boolean) {
+        this.useGlobalJson = useGlobalJson;
         this.packageType = packageType;
+        this.workingDirectory = workingDirectory;
         if (!toolLib.isExplicitVersion(version)) {
             throw tl.loc("ImplicitVersionNotSupported", version);
         }
+        if(!useGlobalJson && (version == null || version.length == 0)){
+            throw tl.loc("VersionMissingException")
+        }
         this.version = version;
         this.cachedToolName = this.packageType === 'runtime' ? 'dncr' : 'dncs';;
     }
 
-    public async install() {
-        // Check cache
-        let toolPath: string;
-        let osSuffixes = this.detectMachineOS();
-        let parts = osSuffixes[0].split("-");
-        this.arch = parts.length > 1 ? parts[1] : "x64";
-        toolPath = this.getLocalTool();
+    private async installFromGlobalJson(osSuffixes: string[]) {
+        let filePathsToGlobalJson = this.getFiles(this.workingDirectory ? this.workingDirectory : ".", "global.json");
+        if(filePathsToGlobalJson.length == 0){
+            throw tl.loc("FailedToFindGlobalJson");
+        }
+        let sdkVersions = new Array<{ name: string, toolPath?: string }>();
+        // read all global files
+        for (let index = 0; index < filePathsToGlobalJson.length; index++) {
+            const filePath = filePathsToGlobalJson[index];
+            let globalJson = {} as { sdk: { version: string } };  
+            console.log("global.json found here: " + filePath);
+            try {
+                globalJson = (JSON.parse(fileSystem.readFileSync(filePath).toString())) as { sdk: { version: string } };
+            } catch (error) {
+                throw tl.error("FailedToReadGlobalJson \r\n" + 
+                "The global.json file isn't valid. (" + filePath + ") Please take a look in the documentation for global.json (https://docs.microsoft.com/en-us/dotnet/core/tools/global-json)");
+            }
+            if(globalJson == null || globalJson.sdk == null || globalJson.sdk.version == null){
+                throw tl.error("FailedToReadGlobalJson \r\n" +  
```

Comment:
same here

Reply:
I think I have resolve this. If not reopen this conversion.